drivers/ipvlan/ipvlan_setup.go

```diff
@@ -0,0 +1,177 @@
+package ipvlan
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/libnetwork/osl"
+	"github.com/vishvananda/netlink"
+)
+
+// Create the ipvlan netlink interface specifying the source name
+func createIPVlan(containerIfName, hostIface, ipVlanMode string) (string, error) {
+	defer osl.InitOSContext()()
+	// Set the ipvlan mode. Default is L2 mode
+	mode, err := setIPVlanMode(ipVlanMode)
+	if err != nil {
+		return "", fmt.Errorf("unsupported %s ipvlan mode: %v", ipVlanMode, err)
+	}
+	// Get the link for the master index (Example: the docker host eth iface)
+	hostEth, err := netlink.LinkByName(hostIface)
+	if err != nil {
+		return "", fmt.Errorf("the requested host interface %s was not found on the Docker host", hostIface)
+	}
+	// Create a ipvlan link
+	ipvlan := &netlink.IPVlan{
+		LinkAttrs: netlink.LinkAttrs{
+			Name:        containerIfName,
+			ParentIndex: hostEth.Attrs().Index,
+		},
+		Mode: mode,
+	}
+	if err := netlink.LinkAdd(ipvlan); err != nil {
+		// verbose but will be an issue if you create a macvlan and ipvlan using the same parent iface
+		logrus.Warnf("Failed to create ipvlan link: %v with the error: %s", ipvlan, err)
+		logrus.Warn("Ensure there are no macvlan links using the same host_interface as the ipvlan iface.")
+		return "", fmt.Errorf("Failed to create the ipvlan link: %v", err)
+	}
+	return ipvlan.Attrs().Name, nil
+}
+
+// setIPVlanMode setter for one of the two ipvlan port types
+func setIPVlanMode(mode string) (netlink.IPVlanMode, error) {
+	switch mode {
+	case modeL2:
+		return netlink.IPVLAN_MODE_L2, nil
+	case modeL3:
+		return netlink.IPVLAN_MODE_L3, nil
+	default:
+		return 0, fmt.Errorf("Unknown ipvlan mode: %s", mode)
+	}
+}
+
+// validateHostIface check if the specified interface exists in the default namespace
+func validateHostIface(ifaceStr string) bool {
+	_, err := net.InterfaceByName(ifaceStr)
```

Comment:
Excellent, I use `net` for iface checks thinking it would be more performant. Great to know, ty.